reservation_units/models/reservation_unit.py

```diff
@@ -268,23 +273,231 @@ class Meta:
-        #  for each reservation unit. Should be replaced with a lookup-property.
-        return ReservationUnitStateHelper.get_state(self)
+    @lookup_property(skip_codegen=True)
+    def active_pricing_type() -> str | None:
+        """Get the pricing type from the currently active pricing."""
+        from reservation_units.models import ReservationUnitPricing
+
+        sq = models.Subquery(
+            queryset=(
+                ReservationUnitPricing.objects.filter(
+                    reservation_unit=models.OuterRef("pk"),
+                    status=PricingStatus.PRICING_STATUS_ACTIVE,
+                )
+                .order_by("begins")
+                .values("pricing_type")[:1]
+            ),
+            output_field=models.CharField(null=True),
+        )
+        return sq  # type: ignore[return-value]  # noqa: RET504
+
+    @active_pricing_type.override
+    def _(self) -> str | None:
+        from reservation_units.models import ReservationUnitPricing
+
+        return (
+            ReservationUnitPricing.objects.filter(
+                reservation_unit=self,
+                status=PricingStatus.PRICING_STATUS_ACTIVE,
+            )
+            .order_by("begins")
+            .values_list("pricing_type", flat=True)
+            .first()
+        )
 
-    @property
-    def reservation_state(self) -> ReservationState:
-        from reservation_units.utils.reservation_unit_reservation_state_helper import (
-            ReservationUnitReservationStateHelper,
+    @lookup_property
+    def publishing_state() -> str:
+        """State indicating the publishing status of the reservation unit."""
+        case = models.Case(
+            # Reservation Unit has been archived.
+            models.When(
+                models.Q(is_archived=True),
+                then=models.Value(ReservationUnitPublishingState.ARCHIVED.value),
+            ),
+            # Reservation Unit is still a draft.
+            models.When(
+                models.Q(is_draft=True),
+                then=models.Value(ReservationUnitPublishingState.DRAFT.value),
+            ),
+            # Reservation Unit is going to be published, but has not been set to unpublish in the future.
+            models.When(
+                (
+                    # Publishes in the future.
+                    models.Q(publish_begins__isnull=False)
+                    & models.Q(publish_begins__gt=Now())
+                    & (
+                        # Does not unpublish.
+                        models.Q(publish_ends__isnull=True)
+                        # Was previously unpublished.
+                        | models.Q(publish_ends__lte=Now())
+                        # Will be unpublished in the future, but before it re-publishes.
+                        | (
+                            models.Q(publish_ends__gt=Now())  #
+                            & models.Q(publish_begins__gt=models.F("publish_ends"))
```

Comment:
This last variant belongs to `PublishingState.SCHEDULED_HIDING` (cf. another comment and `ReservationState.SCHEDULED_RESERVATION`).

Reply:
Done.